doc/1.4/language.md

```diff
@@ -1827,8 +1835,16 @@ C99-style designated initializers are supported for `struct`, `layout`, and
 typedef struct { int x; struct { int i; int j; } y; } struct_t;
 saved struct_t s = { .x = 1, .y = { .i = 2, .j = 3 } }
 ```
-Unlike C, designator lists are not supported, and designated initializers for
-arrays are not supported.
+Unlike C, partial initialization is not allowed implicitly; a designated
+initializer for each member must be specified.
+However, partial initialization can be done explicitly through the use of
+trailing `...` syntax:
+```
+session struct_t s = { .y = { .i = 2, ... }, ... }
+```
+
+Also unlike C, designator lists are not supported, and designated initializers
+for arrays are not supported.
```

Comment:
Feels like we want to write "Initializers follow the same pattern as for `session` variables".
Or extract docs for initializer syntax to a separate section; in the current patch we don't document that partial designated initializers are allowed in e.g. `local` or `return` statements.

Reply:
ugh, you're right. I think we've even said previously that with the increasing complexity to initializers and places we can use them, we're going to want a dedicated section. I feel that's a separate issue, though, deserving of its own JIRA and PR.

Reply:
ok